test/unit/autoplay.test.js

```diff
@@ -29,18 +29,25 @@ QUnit.module('autoplay', {
 
     fixture.appendChild(videoTag);
 
-    // this promise fake will act right away
-    // it will also only act on catch calls
+    // These mock promises immediately execute,
+    // effectively synchronising promise chains for testing
+
+    // This will only act on catch calls
     this.rejectPromise = {
       then(fn) {
         return this;
       },
       catch(fn) {
-        fn();
+        try {
+          fn();
+        } catch (err) {
+          return this;
+        }
         return this;
```

Comment:
Maybe this (line 46) should return `this.resolvePromise` here instead of `this`? I believe that way if a `catch()` throws it will trigger the next `catch`, and if it doesn't throw it will trigger the next `then()`.

Reply:
Returning `this.resolvePromise` here changes the expectations of the `rejectPromise`, this may end up requiring a lot of other test changes that may not be worth it right now.